src/components/App/App.tsx

```diff
@@ -0,0 +1,179 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { useLocation } from 'react-router-dom';
+import { getTodos, postTodo, deleteTodo } from '../../api/todos';
+import { AuthContext } from '../Auth/AuthContext';
+import {
+  Notification, TodoList, Footer, Header,
+} from './index';
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [isNotificationOpen, setIsNotificationOpen] = useState(true);
+  const [textNotification, setTextNotification] = useState<string>('');
+  const [
+    typeNotification,
+    setTypeNotification,
+  ] = useState<'error' | 'success'>();
+  const [isAdding, setIsAdding] = useState<boolean>(false);
+  const [newText, setNewText] = useState<string>('');
+  const [newTodo, setNewTodo] = useState<Todo>();
+  const [deleteCompleted, setDeleteCompleted] = useState<boolean>(false);
+
+  const user = useContext(AuthContext);
+  const location = useLocation();
+
+  const { pathname } = location;
+
+  // для запуска Notification
```

Comment:
Get used to writing comments in English if you want to add any. 
You never know who will work with your code